index.bs

```diff
@@ -4747,6 +4855,9 @@ a numbered step. If outdented, it (today) is rendered as a bullet in the midst o
 1. If [=user verification=] is required for this assertion, verify that the [=User Verified=] bit of the <code>[=flags=]</code> in
     |authData| is set.
 
+1. If the credential [=backup state=] is used as part of Relying Party business logic or policy, compare the previously stored 
```

Comment:
There needs to be advice on what do if the value has changed. Or that at least if the value changes then the RP should make a reasonably policy decision.

Reply:
The use cases are defined. We are not going to make recommendations beyond that. These are hints, not security properties.

Reply:
There is no reference to the word "hint" in this change at all. This means that implementors will read this and WILL believe it IS a security property given it's in a signed part of the attestation/authentication.  This change is completely unclear that it's a hint. This needs correction.

Reply:
PS: There is already evidence that the "hint" about authenticator attachement and UV "hints" have already be assumed to be security properties by RP's which has led to CVE's, so there is established precedent that we MUST be clear about what is a hint, and what is a security property, and this change is currently not achieving that.

Reply:
Sounds like your concern is bigger than this PR. Please create a new issue with your feedback or a PR.

Reply:
No, my concern absolutely matters to this PR. If you are asserting this change is a "hint" and not a security property, then say that explicitly in this PR.

Reply:
I think `{#sctn-credential-backup}}` should be `[[#sctn-credential-backup]]` ?

Reply:
This still doesn't resolve that it is a *HINT* and not a strict property of the device. If this bit is a hint we must call it as such and call it "backup state hint". Otherwise people will assume it is a security property.

Reply:
@timcappalli Could you elaborate on what you mean by the statement "These are hints, not security properties"? Apart from the provision that the state bit may change over time, I don't see any room for the flags to be vague. In what way are they hints (non-compliant and unattested authenticators notwithstanding)?

Reply:
My external assumption related to this situation is that apple passkeys already exist and can be tested and *do not* provide these flags. So the absence or presence of this bit may not be a true reflection of the state (IE apple passkeys don't set the bit today, but are multi device).

Reply:
Why did you make this resolved? You still haven't clarified if this is a hint or a security property? You say it's a hint, but then then spec is written like it's a hard security property. What is it?

Reply:
I'm guessing that the contention is that they are "hard security properties", but under the assumption that the authenticator is compliant. Without attestation there is no guarantee that the authenticator is compliant - and there might not be with attestation either (e.g., Apple passkeys today are not compliant with this proposal, but remember this is still a long time away from proposed standard maturity) - so in that sense they are "hints" because you can't know for sure unless the attestation tells you the authenticator is compliant and trustworthy. So which of the two it is depends only on how much you trust or don't trust the authenticator.

But again, the above is true for the whole spec, not just these flags. The spec generally assumes that authenticators and clients are compliant, but there's no guarantee that they are. Attestation is the only way to get any firm guarantees; without it the only thing you can be completely sure of is that valid assertion signatures can only be generated by an authenticator with possession of the credential private key.

Reply:
If anything, since passkeys from apple are not compliant then the bit flags need to be swapped - 0 means "can be backed up" and we have no idea about the credentials state, and 1 is a hard "yes, we are guaranteed thes credential is not transferable backed up". It's the only way to actually draw a correct view here.

Because if the bit is 0 (IE all devices today) you have to check the attestation to then determine if it's backed up or not, but if it's a 1, then you know the manufacturer is compliant.

Reply:
> but if it's a 1, then you know the manufacturer is compliant.

No you don't. You can only know that if you trust the attestation.

But this did bring up a related point: if we're changing the meaning of 0, then already existing credentials that can be backed up should be allowed to change their backup eligibility (BE) bit from 0 to 1, so we can't say that the BE bit must never change. I'll raise this point in the top-level discussion.

Reply:
> No you don't. You can only know that if you trust the attestation.

I know this. My point is that if the bit is a 1 then the manufacturer is making a token after this documents existing, and is saying "yep, this credential is locked to the device subject to attestation". I'm excruciatingly aware of how this works. 

Regardless, like I said, the question you should be asking is "even with attestation, can an RP assert that this credential is locked to a single cryptographic processor". 

In the current for of this backup state PR the answer is NO an RP can not achieve this. For the primary reason as mentioned with apple. 0 means "it might be backed up" and 1 means "it certainly is". This means that unless apple creates a whole second attestation CA stack in parallel which "guarantees something is not backed up", then any RP has to assume every apple device is a passkey, forever more, meaning they can't be used in pretty much any enterprise RP context. 

So the bit's MUST be swapped, else this PR is practically useless to any RP in a more controlled environment.

Reply:
> Remember as an RP the more important question is "Is this credential locked firmly to this single device" rather than "is this credential backed up or transfered elsewhere?".

This is an overgeneralization. Many RPs will care a lot about the latter question and not at all about the former.

> And in those contexts they can use attestation to validate the manufacturer (IE yubikey with flag 0 is not backed up) where as when apple eventually becomes compliant they can have a 1 in this field.

Right. So you _can_ confirm that a credential is not backed up with the flag as currently proposed. And remember that Apple has no obligation to set the bit to 1 before we reach Proposed Standard maturity with this change.

> Because apple today don't have a seperate attestation cert chain for passkey vs resident key, meaning that in the current form of this proposal, with apple it will be impossible for us to determine if the credential is NOT backed up without apple changing their entire provisioning infrastructure to have two seperate attestation chains.

I'm afraid you'll have to direct that complaint to Apple, it is not a fault of this proposal.